### DISCUSSION TITLE
If only one match found allow clicking on authenticator icon to autofill

### DISCUSSION THREAD

**Discussion by famewolf** [Authenticator-Extension/Authenticator#625] [General]
Right now you have to click on the authenticator icon, click on the numbers of the choice and that will autofill.   It just seems to make more sense if there is only one match to the current domain you are on to default autofill right there.  If more than one show the matches and let the user select.

Either way,  thanks so much for making this extension.  I was so tired of having to go find my phone to enter the code.

**Comment by jordy758** [comment]
This would be awesome.

**Comment by mymindstorm** [comment]
This exact behavior will not be implemented because we don't collect domain info for entries yet, although there is some support for this when overriding the filter. I think this could be more easily achieved by either a dropdown menu or hotkey.

**Comment by famewolf** [comment]
I don't understand the distinction.   You **already** show all possible matches to the site I'm on...I'm just suggesting to run that logic when I click on the extension icon instead of xtension icon then selection if there is only ONE selection.  I have 8 but only one matched the site.  Domain is almost irrelevant.   This would appear to be more a count of the # of entries displayed in the panel and if the count is 1 select that one.  
![2021-02-09_14-53](https://user-images.githubusercontent.com/4558832/107420250-b2387980-6ae6-11eb-865d-b1b6fcd200c6.png)

**Comment by mymindstorm** [comment]
The domain matching currently being used is not secure. Password managers are able to provide this sort of matching because they collect the exact url accounts are associated with. With the current filtering system the extension uses, it would introduce a security risk. For example if you go to `site.com` (legitimate) and `site.net` (evil), then the extension would autofill for both. More on possible issues [here](https://labs.detectify.com/2019/02/28/thinking-outside-of-the-password-manager-box/).

For now, this is effectively blocked until someone has time to redo the filter system.
